windows/src/developer/samples/imsample/imsample.cpp

```diff
@@ -509,7 +524,10 @@ LRESULT CALLBACK IMSampleChildWndProc(HWND hwndChild, UINT msg, WPARAM wParam, L
 
 BOOL FindRule(group *g, rule **rp, WCHAR KeyChar)
 {
-	WCHAR buf[48];
+	if(!g){
+    return FALSE;
+  }
```

Comment:
When is `FindRule` being called with a `nullptr` `g`? That seems like a bug there. And if we're going to check input parameters, we should check `rp` as well:

```suggestion
	if (!g || !rp) {
    return FALSE;
  }
```

Reply:
Yes I can't remember for sure this one was hit or not. Basically, I started out just putting checks on the ones that were crashing but it was slow cycle. add null check; build; fun; crash; view crush dump; repeat.  So then I just went through and added them everywhere.